nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/FetchSlack.java

```diff
@@ -0,0 +1,295 @@
+import javax.json.JsonArray;
+import javax.json.JsonException;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import javax.json.JsonValue;
+import javax.json.JsonValue.ValueType;
+
+import org.apache.http.Header;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicHeader;
+import org.apache.nifi.annotation.behavior.EventDriven;
+import org.apache.nifi.annotation.behavior.InputRequirement;
+import org.apache.nifi.annotation.behavior.InputRequirement.Requirement;
+import org.apache.nifi.annotation.behavior.WritesAttribute;
+import org.apache.nifi.annotation.behavior.WritesAttributes;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.annotation.lifecycle.OnShutdown;
+import org.apache.nifi.annotation.lifecycle.OnUnscheduled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+@Tags({"slack", "download", "fetch", "attachment"})
+@CapabilityDescription("Downloading attachments from slack messages")
+@EventDriven
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@WritesAttributes({
+  @WritesAttribute(attribute = "response.filename", description = "File name parsed from HTTP response."),
+  @WritesAttribute(attribute = "response.content-type", description = "Content-Type parsed from HTTP response."),
+  @WritesAttribute(attribute = "response.content-length", description = "Content-Length parsed from HTTP response."),
+  @WritesAttribute(attribute = "response.date", description = "Date parsed from HTTP response."),
+  @WritesAttribute(attribute = "message.name", description = "File name in the slack JSON."),
+  @WritesAttribute(attribute = "message.mimetype", description = "Mime type in the slack JSON."),
+  @WritesAttribute(attribute = "message.filetype", description = "File type in the slack JSON."),
+  @WritesAttribute(attribute = "message.id", description = "File id in the slack JSON."),
+  @WritesAttribute(attribute = "message.title", description = "File title in the slack JSON."),
+  @WritesAttribute(attribute = "message.size", description = "File size in the slack JSON."),
+  @WritesAttribute(attribute = "message.created", description = "Created timestamp in the slack JSON.")
+})
+public class FetchSlack extends AbstractProcessor {
```

Comment:
We may be able to name this processor better to clarify what it does. How about FetchSlackAttachments? or FetchSlackFiles?